components/admin_console/team_channel_settings/channel/list/channel_list.jsx

```diff
@@ -2,46 +2,143 @@
+    handleChannelSearchKeyUp = async (e) => {
+        const {key} = e;
+        const {searchString} = this.state;
+
+        if (key === Constants.KeyCodes.ENTER[0]) {
+            if (searchString.length > 1) {
+                const response = await this.props.actions.searchAllChannels(searchString, '', false, true, 0, PAGE_SIZE);
+                this.setState({searchMode: true, channels: response.data.channels, searchTotalCount: response.data.total_count, pageResetKey: Date.now()});
+            }
+        }
+
+        if (searchString.length === 0) {
+            this.resetSearch();
+        }
+    }
+
+    getDataBySearch = async (page, perPage) => {
+        const response = await this.props.actions.searchAllChannels(this.state.searchString, '', false, true, page, perPage);
+        const channels = new Array(page * perPage); // Pad the array with empty entries because AbstractList expects to slice the results based on the pagination offset.
+        return channels.concat(response.data.channels);
+    }
+
+    resetSearch = () => {
+        this.setState({searchString: '', channels: [], searchMode: false, searchTotalCount: 0, pageResetKey: Date.now()});
+    }
+
+    header() {
+        return (
+            <>
+                {this.searchBar()}
+                <div className='groups-list--header'>
+                    <div className='group-name adjusted'>
+                        <FormattedMessage
+                            id='admin.channel_settings.channel_list.nameHeader'
+                            defaultMessage='Name'
+                        />
+                    </div>
+                    <div className='group-description'>
+                        <FormattedMessage
+                            id='admin.channel_settings.channel_list.teamHeader'
+                            defaultMessage='Team'
+                        />
+                    </div>
+                    <div className='group-description adjusted'>
+                        <FormattedMessage
+                            id='admin.channel_settings.channel_list.managementHeader'
+                            defaultMessage='Management'
+                        />
+                    </div>
+                    <div className='group-actions'/>
+                </div>
+            </>
+        );
+    }
+
+    onPageChangedCallback = (pagination, channels) => {
+        if (this.state.searchMode) {
+            this.setState({channels});
+        }
+    }
 
-export default class ChannelList extends React.PureComponent {
     render() {
+        const absProps = {...this.props};
+        if (this.state.searchMode) {
+            absProps.actions.getData = this.getDataBySearch;
+        }
         return (
             <AbstractList
-                header={<Header/>}
+                header={this.header()}
                 renderRow={this.renderRow}
-                {...this.props}
+                {...absProps}
+                key={this.state.pageResetKey}
```

Comment:
This `key` prop is being used as per React's recommendation for handling the "reset" of some state:

* https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops
* https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key